projects/pastanaga/src/lib/sidebar/sidebar.component.ts

```diff
@@ -179,14 +179,20 @@ export class SidebarComponent implements OnInit, OnDestroy {
         this.backdrop.classList.add('pa-sidebar-overlay');
         this.renderer.appendChild(this.elementRef.nativeElement.parentElement, this.backdrop);
         this.backdrop.addEventListener('click', () => this.close());
-        this.cdr.markForCheck();
     }
 
     private hideBackdrop() {
         if (!!this.backdrop && !!this.backdrop.parentNode) {
             this.backdrop.parentNode.removeChild(this.backdrop);
             this.backdrop = null;
-            this.cdr.markForCheck();
         }
     }
+
+    private markForCheck() {
+        window.setTimeout(() => {
```

Comment:
Why do we need to do this `setTimeout`? It's like we are doing something wrong :thinking:

Reply:
agree with Luis, it feels unsafe to encapsulate a `markForCheck` in a timeout. so timeout is not working outside zone, so I guess it's ok to keep `markForCheck` and don't use `detectChanges`, but still, why would you have it always in a timeout like that?

Reply:
I feel the same, but if we don't do that, and if the parent component is doing a `markForCheck` on its side, then we get some Angular view check errors a way or another.

I spent the afternoon on that and could not come up with a better solution.

So I was planning to merge it (and if you have some time, you can try to find a better implementation).